src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadsTest.java

```diff
@@ -1236,5 +1236,110 @@ public void testCisco_CallManager() {
         Assert.assertTrue(report.getSsaEnabled());
     }
 
+    @Test
+    public void testOracleJDK8OnLinux() {
+        Map<String, Object> facts = new HashMap<>();
+        // always add a String fact with the name of the agenda group defined in the DRL file (e.g. "SourceCosts")
+        facts.put("agendaGroup", "Workloads");
+
+        VMWorkloadInventoryModel vmWorkloadInventoryModel = new VMWorkloadInventoryModel();
+        Map<String, String> files = new HashMap<>();
+        files.put("/usr/java/latest/release", "JAVA_VERSION=\\\"1.8.11");
+        vmWorkloadInventoryModel.setFiles(files);
+        List<String> systemServicesNames = new ArrayList<>();
+        systemServicesNames.add("NOTwas");
+        vmWorkloadInventoryModel.setSystemServicesNames(systemServicesNames);
+        facts.put("vmWorkloadInventoryModel", vmWorkloadInventoryModel);
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+
+        Assert.assertEquals(3, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener, "AgendaFocusForTest", "SsaEnabled_System_Services_Present", "Workloads_Oracle_JDK_8_On_Linux");
+
+        List<WorkloadInventoryReportModel> reports = Utils.extractModels(GET_OBJECTS_KEY, results, WorkloadInventoryReportModel.class);
+
+        // just one report has to be created
+        Assert.assertEquals(1, reports.size());
+        WorkloadInventoryReportModel report = reports.get(0);
+        Assert.assertNotNull(report.getWorkloads());
+        Assert.assertEquals(1, report.getWorkloads().size());
+        Assert.assertTrue(report.getWorkloads().stream().anyMatch(workload -> workload.toLowerCase().contains("Oracle JDK 8".toLowerCase())));
+        Assert.assertTrue(report.getSsaEnabled());
+    }
+
+    @Test
+    public void testOracleJDK11OnLinux() {
```

Comment:
all 3 tests are 99% identical ..... probably you could extract the common code and refactor them ? to remove duplicated code